src/StoreApi/Routes/V1/AI/Product.php

```diff
@@ -81,6 +81,7 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 		}
 
 		$product_updater->update_product_content( $product_information );
+		flush_rewrite_rules();
```

Comment:
Unfortunately, this solution won't work, as the Product endpoint is called for updating each one of the products individually, so we are still flushing rewrite rules multiple times.

To circumvent this, we can do a combined change on the client side and on the Product endpoint: I hope you don't mind I'm pushing change to this PR here to expedite things: a new `'last_product'` param is being added to the Product endpoint & also I'm opening a PR on core to update `services.ts` with `last_product: index === response.product_content.length,`:

```
		const productContents = response.product_content.map(
			( product, index ) => {
				return apiFetch( {
					path: '/wc/private/ai/product',
					method: 'POST',
					data: {
						products_information: product,
						last_product: index === response.product_content.length,
					},
				} );
			}
		);
```